File: programmers/level1/x만큼_간격이_있는_n개의_숫자/10003_x만큼간격이있는n개의숫자.java
class Solution {
    public long[] solution(int x, int n) {
        long[] answer = new long[n];
        long temp = 0;
        for (int i=0; i<n ; i++) {
            temp += x;
            answer[i] = temp;
            System.out.println(temp);
        }
        
        return answer;
    }
}

// 가장 인상깊은 코드 
// 공식을 사용하여 변수 선언을 최소화하여 코드를 간결하게 만든것이 인상깊었다.
//
// class Solution {
//   public long[] solution(long x, int n) {
//       long[] answer = new long[n];
//       for(int i = 0; i < n; i++){
//           answer[i] = x * (i + 1);
//       }
//       return answer;
//   }
// }

File: programmers/level1/x만큼_간격이_있는_n개의_숫자/10003_x만큼간격이있는n개의숫자.js
function solution(x, n) {
    var answer = [];
    let temp = 0; 
    for (let i=0 ; i<n ; i++) {
        temp += x;
        answer.push(temp);
    };
    
    return answer;
}

// 가장 인상깊은 코드 
// 길이가 n인 리스트를 생성하고 모든 indec에 x값을 넣어둔뒤 map을 통해 index * (x+1)로 재배열한다.
//
// function solution(x, n) {
//     return Array(n).fill(x).map((v, i) => (i + 1) * v)
// }
